optuna/visualization/matplotlib/_slice.py

```diff
@@ -1,11 +1,152 @@
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+        params:
+            Parameter list to visualize. The default is all parameters.
+
+    Returns:
+        A :class:`matplotlib.axes.Axes` object.
+    """
+
+    _imports.check()
+    return _get_slice_plot(study, params)
+
+
+def _get_slice_plot(study: Study, params: Optional[List[str]] = None) -> "Axes":
+
+    # Calculate basic numbers for plotting.
+    trials = [trial for trial in study.trials if trial.state == TrialState.COMPLETE]
+
+    if len(trials) == 0:
+        _logger.warning("Your study does not have any completed trials.")
+        _, ax = plt.subplots()
+        return ax
+
+    all_params = {p_name for t in trials for p_name in t.params.keys()}
+    if params is None:
+        sorted_params = sorted(list(all_params))
+    else:
+        for input_p_name in params:
+            if input_p_name not in all_params:
+                raise ValueError("Parameter {} does not exist in your study.".format(input_p_name))
+        sorted_params = sorted(list(set(params)))
+
+    n_params = len(sorted_params)
+
+    # Set up the graph style.
+    cmap = plt.get_cmap("Blues")
+    padding_ratio = 0.05
+    plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
+
+    # Prepare data.
+    obj_values = [t.value for t in trials]
+
+    if n_params == 1:
+        # Set up the graph style.
+        fig, axs = plt.subplots()
+        axs.set_title("Slice Plot")
+
+        # Draw a scatter plot.
+        sc = _generate_slice_subplot(
+            trials, sorted_params[0], axs, cmap, padding_ratio, obj_values  # type: ignore
+        )
+    else:
+        # Set up the graph style.
+        min_figwidth = matplotlib.rcParams["figure.figsize"][0] / 2
+        fighight = matplotlib.rcParams["figure.figsize"][1]
+        # Ensure that each subplot has a minimum width without relying on auto-sizing.
+        fig, axs = plt.subplots(
+            1, n_params, sharey=True, figsize=(min_figwidth * n_params, fighight)
+        )
+        fig.suptitle("Slice Plot")
+
+        # Draw scatter plots.
+        for i, param in enumerate(sorted_params):
+            ax = axs[i]
+            sc = _generate_slice_subplot(
+                trials, param, ax, cmap, padding_ratio, obj_values  # type: ignore
+            )
+
+    axcb = fig.colorbar(sc, ax=axs)
+    axcb.set_label("#Trials")
+
+    return axs
+
+
+def _generate_slice_subplot(
+    trials: List[FrozenTrial],
+    param: str,
+    ax: "Axes",
+    cmap: "Colormap",
+    padding_ratio: float,
+    obj_values: List[Union[int, float]],
+) -> "PathCollection":
+    x_values = []
+    y_values = []
+    trial_numbers = []
+    for t in trials:
+        if param in t.params:
+            x_values.append(t.params[param])
+            y_values.append(obj_values[t.number])
```

Comment:
If some trials are pruned, `t.number` and the index of `t` in the `trials` would be different.

```suggestion
    for i. t in enumerate(trials):
        if param in t.params:
            x_values.append(t.params[param])
            y_values.append(obj_values[i])
```

Reply:
Thank you for reviewing! I agree with you.